docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,261 @@
+- Tailwind classes can be converted into inline styles to ensure maximum compatibility of our emails.
+- It is highly configurable leading to the possibility of reusing the config file setup in `fxa-settings` for our templates
+- With tailwind's utility and compononent classes, we can DRY up our codebase
+
+### Email Previewing
+
+Chosen option: "Option B - Use Storybook", because:
+
+- It is easy to setup and provides us the flexibility to preview all of the different email states together.
+- We have a Storybook deployment already in place and hooking it up for `fxa-auth-server` will ensure consistency.
+- Easy to test and perform QA without needing to touch the codebase.
+- Support for TailwindCSS
+
+Note: Along with using storybook to view the various states of the templates, we are also planning on continued support of [Maildev](https://www.npmjs.com/package/maildev) since it has applications beyond just previewing emails.
+
+### Other
+
+**Plaintext files**: We'll be converting our React templates over to HTML and once the HTML templates are generated we can extract the text out of them using a HTML parser like [html-to-text](https://github.com/html-to-text/node-html-to-text) or [textversionjs](https://github.com/EDMdesigner/textversionjs)
+
+**Localization**: We will upgrade the localization tool from GetText to [Fluent](https://www.npmjs.com/package/@fluent/react) since it's preferred by the l10n team and other FxA front-ends are using it.
+
+**Integration & QA**: During development templates will be marked as being part of a release group. This could be in the form of a mapped list of template names or some variable associated with the template file. Each group will have a corresponding environment variable flag. When this release group’s flag is enabled any templates that fall under it will be used when generating an email template in that environment; until a template's corresponding release group is enabled it is not used and the old/current template will continue to be served. Release groups will be initially enabled in staging until QA has had an opportunity to thoroughly test and evaluate each template in the group, after which they can be incrementally enabled in production.
+
+## Pros and Cons of the Templating Options
+
+`fxa-auth-server` currently makes use of Mustache to develop email templates. Mustache is a logicless HTML templating framework and has been proven to be super helpful in developing mail components so far, but due to its logicless nature it becomes tedious to include conditional statements in the mail components. The following section will detail upon the various pros and cons of potential libraries/ frameworks that could be used to replace Mustache as a templating solution.
+
+### Option A - Continue to use Mustache
+
+- Description
+  - Keep the email templates as is in Mustache
+- Pros
+  - The effort to refactor all the templates would be eliminated for now
```

Comment:
Mustache is very easy to read, so the simplicity and ease of use could be considered a Pro.

Reply:
Maybe personal preference, but: When I was working on the email templates, I found Handlebars was *too* simple.

Or, at least, it complicated things insofar as the display logic had to be split between the backend and the template. That, or [we need to add](https://github.com/mozilla/fxa/commit/fe383495d69d7aa7458879129fd84beef50b7063#diff-8ca895f034812adeed032afe4311f65cc0e97191828ddea8c7d0ec7e26d48ad6R34-R61) [our own helpers](https://github.com/mozilla/fxa/commit/d6ad532434676100994fbe7e47d1e4e62a9d1a1e) to expand it ad hoc.

I tend to like a system where the backend can just pass in a bag of data and the template can handle it from there. EJS would get us there, so would React.